src/Proof.purs

```diff
@@ -77,15 +86,28 @@ type ProofRow
     }
 
 type Box
-  = { startIdx :: Int
-    -- TODO Store accessible sub-boxes for →i/BottomElim
+  = Tuple Int Int
+
+-- | A scope which contains the lines, boxes and vars available.
+type Scope
+  = { lines :: Array Int
+    , boxes :: Array Box
+    , vars :: Array Variable
```

Comment:
Rätta mig om jag har fel men `vars` kommer ej behövas då den informationen redan trådas genom regelappliceringarna

Reply:
Jag ser inte riktigt hur man löser forall-intro utan att ha `vars` i `Scope`. Om man kollar de två bevisen på sidan 119 i "Logic in Computer Science" så ser jag inte hur man validerar dem utan att ha x<sub>0</sub> i `Scope`.

Har du en bra lösning på det så förklara gärna.

Reply:
Ser inte hur det blir problem för de exemplena då forall-intro kan kolla upp x₀ som första fresh raden i boxargumentet. Däremot inser jag nu att det kan bli problem om man får sätta vilken variabel som helst som fresh, t.ex. här

```
1. P(x₀)      Premise
2. | x₀       Fresh
3. | P(x₀)    Copy 1
4. ∀x P(x)    ∀i 2-3
```

där rad två borde bli ogiltlig.